rust/src/dns/parser.rs

```diff
@@ -361,16 +361,45 @@ pub fn dns_parse_body<'a>(
 ) -> IResult<&'a [u8], DNSMessage> {
     let (i, queries) = count(|b| dns_parse_query(b, message), header.questions as usize)(i)?;
     let (i, answers) = dns_parse_answer(i, message, header.answer_rr as usize)?;
-    let (i, authorities) = dns_parse_answer(i, message, header.authority_rr as usize)?;
-    let (i, additionals) = dns_parse_answer(i, message, header.additional_rr as usize)?;
+
+    let mut invalid_authorities = false;
+    let mut authorities = Vec::new();
+    let mut i_next = i;
+    let authorities_parsed = dns_parse_answer(i, message, header.authority_rr as usize);
+    match authorities_parsed {
+        Ok((i, authorities_ok)) => {
+            authorities = authorities_ok;
+            i_next = i;
+        }
+        _ =>{
+            invalid_authorities = true;
+        }
+    }
```

Comment:
nit: `if let Ok` would look better to the eye.

Reply:
Indeed, will do